core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -719,10 +719,23 @@ class SparkContext(config: SparkConf) extends Logging {
     if (value == null) {
       localProperties.get.remove(key)
     } else {
+      checkLocalProperty(key, value)
```

Comment:
Do we expect users would set `spark.job.interruptOnCance` as I notice it's private? I think it may be better to prevent users from setting this kind of private property instead of doing check valid check.  And we could recommend users to use `setJobGroup` instead.

Reply:
I perfer to the suggestion, forbid is better. If so we might need an another internal method to set local property for some test case.

Reply:
Yeah, we need the internal method if we want to forbid. May let's see others' feedback first.

Reply:
we should just change this to use the ConfigBuilder. mark it as internal and the toBoolean function already does this same check to make sure valid.  That doesn't prevent them from doing it though so if we wanted to add other checking for all internal that might be a bit more work and complex as I'm sure we use some for testing.

Reply:
oh sorry I guess we can't do that because local property